src/pages/index.tsx

```diff
@@ -64,9 +22,17 @@ const AuthShowcase: React.FC = () => {
 
   const { data: secretMessage } = api.example.getSecretMessage.useQuery(
     undefined, // no input
-    { enabled: sessionData?.user !== undefined },
+    { enabled: sessionData?.user !== undefined }
   );
 
+  const isAdmin = useAdminCheck();
+
+  useEffect(() => {
+    if (isAdmin) {
+      location.href = "/admin";
```

Comment:
change to use `router.push('/admin')`

Reply:
Actually, I dont think you need an effect at all in this case. Consider replacing the entire `useEffect` with the code below
```
  const router = useRouter();

  if (isAdmin) {
    void router.replace("/admin");
  }

```